### PR TITLE
Add key_to_character function

### PR DESCRIPTION
**Pull request by willmcgugan**
Fixes https://github.com/Textualize/textual/issues/4944 and https://github.com/Textualize/textual/issues/4930

### CODE REVIEW

File: src/textual/dom.py [hunk 1]
```diff
@@ -1037,6 +1037,7 @@ def check_consume_key(self, key: str) -> bool:
 
         Args:
             key: A key identifier.
+            character: A character associated with they key, or `None` if there isn't one.
```
Comment (TomJGooding) [inline src/textual/dom.py:1040]:
```suggestion
            character: A character associated with the key, or `None` if there isn't one.
```

File: src/textual/widgets/_input.py [hunk 2]
```diff
@@ -361,18 +361,19 @@ def _cursor_at_end(self) -> bool:
...
 
-    def check_consume_key(self, key: str) -> bool:
+    def check_consume_key(self, key: str, character: str | None) -> bool:
         """Check if the widget may consume the given key.
 
         As an input we are expecting to capture printable keys.
 
         Args:
             key: A key identifier.
+            character: A character associated with they key, or `None` if there isn't one.
```
Comment (TomJGooding) [inline src/textual/widgets/_input.py:371]:
```suggestion
            character: A character associated with the key, or `None` if there isn't one.
```

File: src/textual/widgets/_text_area.py [hunk 3]
```diff
@@ -546,22 +546,26 @@ def _get_builtin_highlight_query(language_name: str) -> str:
...
 
-    def check_consume_key(self, key: str) -> bool:
+    def check_consume_key(self, key: str, character: str | None = None) -> bool:
         """Check if the widget may consume the given key.
 
         As a textarea we are expecting to capture printable keys.
 
         Args:
             key: A key identifier.
+            character: A character associated with they key, or `None` if there isn't one.
```
Comment (TomJGooding) [inline src/textual/widgets/_text_area.py:556]:
```suggestion
            character: A character associated with the key, or `None` if there isn't one.
```

**Review by TomJGooding — COMMENTED** [review]
Sorry, hope you don't mind me sticking my nose in but couldn't help spotting this while in spell check mode!

**Review by darrenburns — APPROVED** [review]

**Pull request merged**